### PR TITLE
[SPARK-34375][CORE][K8S][TEST] Replaces 'Mockito.initMocks' with 'Mockito.openMocks'

### PR DESCRIPTION
**Pull request by LuciferYang**
### What changes were proposed in this pull request?
`Mockito.initMocks(Object)` is a deprecated api, should use `Mockito.openMocks(Object).close()` instead.

### Why are the changes needed?
Cleanup deprecation api usage.

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
Pass the Jenkins or GitHub Action

### CODE REVIEW

File: core/src/test/java/org/apache/spark/shuffle/sort/UnsafeShuffleWriterSuite.java [hunk 1]
```diff
@@ -98,7 +98,7 @@ public void tearDown() {
   @Before
   @SuppressWarnings("unchecked")
   public void setUp() throws IOException {
-    MockitoAnnotations.initMocks(this);
+    MockitoAnnotations.openMocks(this);
```
Comment (HyukjinKwon) [inline core/src/test/java/org/apache/spark/shuffle/sort/UnsafeShuffleWriterSuite.java:101]: Shouldn't we do `openMocks(this).close()`? that's what the docs say.

Comment (LuciferYang) [reply core/src/test/java/org/apache/spark/shuffle/sort/UnsafeShuffleWriterSuite.java:101]: After tracing the code, I found that the `close()` method alway `do nothing`, but I'll change them all to `openMocks(this).close()` later :)

Comment (LuciferYang) [reply core/src/test/java/org/apache/spark/shuffle/sort/UnsafeShuffleWriterSuite.java]: a8ebb43 fix these

**Review by HyukjinKwon — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
Merged to master.

**Comment by LuciferYang** [comment]
thx ~ @HyukjinKwon

**Pull request closed without merging**